metrics/handler.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog"
+)
+
+// HttpHandler is a thin middleware for gathering metrics about http request.
```

Comment:
I'm not sure of how correct it is. I want this handler to wrap the default one to measure runtime properly. However, if there's no `method` field in a request I have to silently not gather any metrics. 

Is it fine? If no, what would you recommend to do?

Reply:
I don't want to write BadRequest to a response write in case there's no `method` field because we will have 2 places in the code base where we handle errors. I'd rather keep all error handling in `server.go` handler.

Reply:
if you can't parse the method then the request is invalid and you are correct to not handle it, you should even just write a debug output not info, because these requests are not really relevant. We should at some later point refactor this parsing of request to a single place instead, if in case the geth implementation doesn't have it already. I think it would be good to stop relying on geth implementation for server in the first place, because I don't like how they made it, it makes extending things like these very hard. We will likely refactor that in the future. So this is a bit hacky approach because we rely a lot on geth server.